converters/toarray.ts

```diff
@@ -0,0 +1,47 @@
+import { IEvent, IEventArray, ILayoutState } from "../clarity";
+import { getNodeIndex } from "./../src/plugins/layout/stateprovider";
+import { hashCode } from "./../src/utils";
+import schemas from "./schema";
+
+// We serialize send a lot of JSONs with identical structures and putting all property names on the
+// wire every time can be very redundant. We can save bytes by splitting JSON into a separate nested array of
+// its property values and a separate nested array of its property names (schema). This way, each unique schema
+// has to only be sent for the first event of that type and the following events can then re-use this schema
+// on the server for reconstructing the full JSON.
+export default function(event: IEvent): IEventArray {
+  let keys = Object.keys(event);
+  let schema = schemas.createSchema(event.state);
+  let newSchema = schemas.addSchema(schema);
```

Comment:
Combine into one function in the schemas module.

Reply:
I'm getting two data pieces here - a schema itself and a boolean indicating whether it's a new schema or not. I don't want to return combine them into one return value as an array or some custom object.